src/mds/Server.cc

```diff
@@ -526,7 +526,8 @@ void Server::handle_client_reclaim(const cref_t<MClientReclaim> &m)
     return;
   }
 
-  if (m->get_flags() & MClientReclaim::FLAG_FINISH) {
+  //let reclaim_session handle negative reclaim flags
+  if (((int32_t) m->get_flags()) > 0 && (m->get_flags() & MClientReclaim::FLAG_FINISH)) {
```

Comment:
I would consider it an error if the client specifies `FLAG_FINISH` and other flags too.

Reply:
I didn't modify the way it looks for the flag, do you mean `if (m->get_flags() & MClientReclaim::FLAG_FINISH) {` is not the correct way to do it? Do you want me to fix this?

Reply:
@batrick just curious, `if (m->get_flags() & MClientReclaim::FLAG_FINISH)` doesn't actually mean that client is specifying FLAG_FINISH and other flags at the same time right? I think it is trying to match the bits of the flag it received against `MClientReclaim::FLAG_FINISH` so that if it returns 1 then `Server::reclaim_session` is done and it's safe to do `Server::finish_reclaim_session` straightaway else we do try `Server::reclaim_session` ?